### PR TITLE
Class testing

### PR DESCRIPTION
**Pull request by laurenlucero**
# Pull Request
## Please include a summary of the change:
I created my mock data files for guests, rooms and bookings.
I made classes for Guest, Room, Booking, Manager and Hotel.
Room and Booking currently holding properties only.
Guest, Manager and Hotel have most of the methods and testing completed.
Need to review all classes against spec to see if additional functionality is needed.
I also added to basic structure and styling.

## If it is a new feature, describe the feature and the iteration of the project spec it addresses. Note any questions or struggles as well.
These features address iterations 1 and 2. 
Manager can login for iteration 1 but need to refactor so that guests can login as well as DOM updates.
Iteration 2 methods have been addressed in classes and testing.
Additional methods from next iterations may be needed.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Files Modified:
	modified:   src/Guest.js
	modified:   src/Hotel.js
	modified:   src/css/base.scss
	modified:   src/domUpdates.js
	modified:   src/index.html
	modified:   src/index.js
	modified:   test/Guest-test.js
	modified:   test/Hotel-test.js
	modified:   test/mock-booking-data.js
	deleted:    test/sample-test-data.js

### CODE REVIEW

**Comment by laurenlucero** [comment]
@Karlfunhouse PR buddy please review and merge if no conflix or concerns
Instructors @hannahhch @thatpamiam

File: src/Guest.js [hunk 1]
```diff
@@ -1,28 +1,64 @@
+// select a room for booking
+// select a date to book a room
+
 class Guest {
-  constructor(id, name) {
-    this.id = id;
-    this.name = name;
+  constructor(guestData, bookingData) {
```
Comment (Karlfunhouse) [inline src/Guest.js:5]: I think this looks like a great data flow

File: src/Guest.js [hunk 2]
```diff
@@ -1,28 +1,64 @@
...
+    // find any bookings past/present or upcoming
+    let guestBookings = this.bookings.filter(
+      booking => booking.userID === this.id
+    );
+    return guestBookings;
   }
 
-  filterAvailableRooms() {
-    // filter list of available rooms by roomType property
+  calculateTotalSpending(roomData) {
```
Comment (Karlfunhouse) [inline src/Guest.js:26]: Nice nested prototypes

File: src/Guest.js [hunk 3]
```diff
@@ -1,28 +1,64 @@
...
+      .reduce((total, cost) => {
+        total += cost;
+        return total;
+      }, 0);
   }
 
+// TBT (To Be Tested) bookRoom method
   bookRoom() {
-    // select a room for booking
+    let bookingObject = {
```
Comment (Karlfunhouse) [inline src/Guest.js:46]: This also looks awesome nice work

File: src/Manager.js [hunk 4]
```diff
@@ -1,19 +1,87 @@
...
+      },
+      body: JSON.stringify(bookingObject)
+    })
+      .then(response => console.log(response.json()))
+      .catch(err => console.log(err.message));
   }
 
-  deleteBooking() {
-    // Delete any upcoming room bookings for a user
+  // TBT
```
Comment (Karlfunhouse) [inline src/Manager.js:73]: Have you gotten this to work?  I haven't gotten to this part yet so I'm curious!

File: src/index.html [hunk 5]
```diff
@@ -2,23 +2,98 @@
...
+        <h3>Rooms Occupied</h3>
+        <p class="rooms-occupied"></p>
+      </div>
+
+      <div class="dashboard-section">
+        <h3>Revenue</h3>
+        <p class="revenue"></p>
+      </div>
+
+      <div class="dashboard-section">
```
Comment (Karlfunhouse) [inline src/index.html:69]: This also looks great!

**Pull request merged**